unstructured_ingest/v2/processes/connectors/vectara.py

```diff
@@ -0,0 +1,323 @@
+import asyncio
+import json
+import uuid
+from dataclasses import dataclass, field
+from datetime import date, datetime
+from pathlib import Path
+from typing import Any, Dict, List, Mapping, Optional
+
+from dateutil import parser
+from pydantic import Field, Secret
+
+from unstructured_ingest.error import DestinationConnectionError
+from unstructured_ingest.utils.data_prep import flatten_dict
+from unstructured_ingest.utils.dep_check import requires_dependencies
+from unstructured_ingest.v2.interfaces import (
+    AccessConfig,
+    ConnectionConfig,
+    FileData,
+    Uploader,
+    UploaderConfig,
+    UploadStager,
+    UploadStagerConfig,
+)
+from unstructured_ingest.v2.logger import logger
+from unstructured_ingest.v2.processes.connector_registry import DestinationRegistryEntry
+
+BASE_URL = "https://api.vectara.io/v1"
+
+CONNECTOR_TYPE = "vectara"
+
+
+class VectaraAccessConfig(AccessConfig):
+    oauth_client_id: Optional[str] = Field(default=None, sensitive=True, description="Client ID")
+    oauth_secret: Optional[str] = Field(default=None, sensitive=True, description="Client Secret")
+
+
+class VectaraConnectionConfig(ConnectionConfig):
+    access_config: Secret[VectaraAccessConfig]
```

Comment:
Are either of the values in the access config required? I see they're typed with `Optional`. If so, we can set this to an empty model by default here (look at pinecone).